server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/AliasRoutesTest.java

```diff
@@ -0,0 +1,275 @@
+import org.apache.james.domainlist.lib.DomainListConfiguration;
+import org.apache.james.domainlist.memory.MemoryDomainList;
+import org.apache.james.metrics.logger.DefaultMetricFactory;
+import org.apache.james.rrt.api.RecipientRewriteTable;
+import org.apache.james.rrt.api.RecipientRewriteTableException;
+import org.apache.james.rrt.lib.MappingSource;
+import org.apache.james.rrt.memory.MemoryRecipientRewriteTable;
+import org.apache.james.user.api.UsersRepository;
+import org.apache.james.user.memory.MemoryUsersRepository;
+import org.apache.james.webadmin.WebAdminServer;
+import org.apache.james.webadmin.WebAdminUtils;
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.util.Map;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.when;
+import static org.apache.james.webadmin.Constants.SEPARATOR;
+import static org.apache.james.webadmin.WebAdminServer.NO_CONFIGURATION;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.CoreMatchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+
+class AliasRoutesTest {
+
+    private static final Domain DOMAIN = Domain.of("b.com");
+    public static final String BOB = "bob@" + DOMAIN.name();
+    public static final String BOB_WITH_SLASH = "bob/@" + DOMAIN.name();
+    public static final String BOB_WITH_ENCODED_SLASH = "bob%2F@" + DOMAIN.name();
+    public static final String BOB_ALIAS = "bob-alias@" + DOMAIN.name();
+    public static final String BOB_ALIAS_WITH_SLASH = "bob-alias/@" + DOMAIN.name();
+    public static final String BOB_ALIAS_WITH_ENCODED_SLASH = "bob-alias%2F@" + DOMAIN.name();
+    public static final String ALICE = "alice@" + DOMAIN.name();
+    public static final String BOB_PASSWORD = "123456";
+    public static final String BOB_WITH_SLASH_PASSWORD = "abcdef";
+    public static final String ALICE_PASSWORD = "789123";
+
+    private WebAdminServer webAdminServer;
+
+    private void createServer(AliasRoutes aliasRoutes) throws Exception {
+        webAdminServer = WebAdminUtils.createWebAdminServer(
+            new DefaultMetricFactory(),
+            aliasRoutes);
+        webAdminServer.configure(NO_CONFIGURATION);
+        webAdminServer.await();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+            .setBasePath("address/aliases")
+            .log(LogDetail.METHOD)
+            .build();
+    }
+
+    @AfterEach
+    void stop() {
+        webAdminServer.destroy();
+    }
+
+    @Nested
+    class NormalBehaviour {
```

Comment:
Can we have tests checking that the record was added in the RRT ?

Reply:
I believe I would need the GET methods for that, that should come in a later PR?

Reply:
Can we get a simple test testing the underlying structure as a temporary workaround?

Shipping untested code is bad...